### DISCUSSION TITLE
Album cover does not appear in Android Application

### DISCUSSION THREAD

**Discussion by fsan2val** [AzuraCast/AzuraCast#6862] [Q&A]
Hello

My app developer says that the image cannot be displayed for this reason.

_It has nothing to do with http or https, the most important thing is the metadata displayed, it should be noted that the web version and the Android version are different, we use the exoplayer library as streaming media, and the metadata displayed is in accordance with what was detected by the exoplayer library, that is absolute and cannot be changed. Then to get the album image, we use the iTunes API which is based on the metadata name displayed. So if the image does not appear, then the image search results will not provide results related to the metadata name._

What do you recommend please?

Thank you

![Screenshot-12-ene-2024-14-01-58](https://github.com/AzuraCast/AzuraCast/assets/69484466/cd8fcda6-c4c6-4e35-88dd-b311e69e940f)

**Comment by BusterNeece** [comment]
We have a full suite of APIs for providing the rich metadata that we maintain inside AzuraCast alongside your audio: https://www.azuracast.com/docs/developers/now-playing-data/

If your developer can use those, you'll have the ability to fetch the album artwork that we have internal to our application (including any remote artwork we fetch through our own services), along with a full host of other relevant data that can't be communicated in the tiny metadata segment that's inside the audio stream itself.

**Reply by fsan2val** [reply]
The developer only uses the streaming transmission link.

It should be noted that I have used another transmission link from another platform with icecast (Not Azuracast) as a test and it works very well.

**Reply by BusterNeece** [reply]
Do you have a live DJ connecting to the stream? It's a known issue that the initial metadata for a live DJ when they initially connect often doesn't go through, but the second update will go through just fine. We're not modifying the metadata in such a way that it would look like that out of the box unless that's how a live DJ was sending the metadata.

**Reply by fsan2val** [reply]
I'm not using live DJ for now, just scheduled Auto DJ from azuracast.

In the public link, if the album cover appears, it is only when I use the link and apply third-party applications such as a play or an app, the cover does not appear there.

Just in case I only use icecast on the platform

**Reply by BusterNeece** [reply]
Well my point remains, if you're just playing regular music then AzuraCast will send the metadata for those tracks through just fine, it won't just send something like the station name instead. I'd recommend that if you want to take advantage of our internal album art and other rich metadata support, that you use our APIs, but otherwise I don't have much advice to give on this front.

**Reply by fsan2val** [reply]
In fact there shouldn't be a problem, but there are. I attach an image where you can see the official play of azuracast.

What I would like is that only with the transmission link. I don't know if it only happens to me or to other people? If you can share a transmission link to test, maybe it is my installation, I say.

![Radio-27-Dance](https://github.com/AzuraCast/AzuraCast/assets/69484466/052bf269-bd3b-41b6-8f13-a5a68942a137)

**Reply by BusterNeece** [reply]
@fsan2val That *has* to be in the metadata for that track. We don't add station names or anything like that to media we play without it being set in the media's metadata. That or you're using some custom variation of the software that changes that.

**Reply by fsan2val** [reply]
The artist name and song metadata work fine, the image is just an example to tell you that the album cover is also visible, something that I still can't get to be seen in the android application. I hope there is a miracle

**Reply by BusterNeece** [reply]
Well, there's no miracle, that album art is provided in our own APIs that I linked to above. Your developers can use those APIs. If they're just feeding off of Icecast's metadata, all they'll ever get is "Title - Artist", and there's no way they can extrapolate that to the artwork you set inside AzuraCast.

For them not to use our simple JSON API for this purpose would be laziness on their part.

**Reply by fsan2val** [reply]
Perfectly, I will recommend to the developer what he mentions. You have a section for managing the API. For now, very grateful for your guidance.
